### PR TITLE
Fixing the 'setup a code' documentation page

### LINKED ISSUE #3015: Check documentation section on code setup
Check if it the section is updated to AiiDA 1.0, if it is correct, and if remove possible redundancies

https://aiida.readthedocs.io/projects/aiida-core/en/latest/get_started/codes.html

### PR DESCRIPTION
**Pull request by d-tomerini**
Fixes #3015 

Adapted the documentation page to reflect the changes introduced in AiiDA 1.0

Specifically:
- Some keys changed name, plugin list command
- Pre and Post execution lines
- Not allowed to set 'None' as calculation input plugin

### CODE REVIEW

File: docs/source/get_started/codes.rst [hunk 1]
```diff
@@ -55,69 +55,58 @@ To setup a new code, you execute::
...
-* **default input plugin**: A string that identifies the default input plugin to
+* **Default calculation input plugin**: A string that identifies the default input plugin to
   be used to generate new calculations to use with this code.
   This string has to be a valid string recognized by the ``CalculationFactory``
   function. To get the list of all available Calculation plugin strings,
-  use the ``verdi calculation plugins`` command. Note: if you do not want to 
-  specify a default input plugin, you can write the string "None", but this is
-  strongly discouraged, because then you will not be able to use
-  the ``.get_builder`` method of the ``Code`` object.
+  use the ``verdi plugin list aiida.calculations`` command.
```
Comment (borellim) [inline docs/source/get_started/codes.rst]:
```suggestion
  use the ``verdi plugin list aiida.calculations`` command.
```

File: docs/source/get_started/codes.rst [hunk 2]
```diff
@@ -55,69 +55,58 @@ To setup a new code, you execute::
...
-    * **Folder with the code**: The folder on your local computer in which there
-      are the files to be stored in the AiiDA repository, and that will then be
-      copied over to the remote computers for every submitted calculation.
-      This must be an absolute path on your computer.
-    * **Relative path of the executable**: The relative path of the executable
-      file inside the folder entered in the previous step.
-  
   * REMOTE CODES:
   
     * **Remote computer name**: The computer name as on which the code resides,
```
Comment (borellim) [inline docs/source/get_started/codes.rst]:
```suggestion
     * **Remote computer name**: The computer name on which the code resides,
```

File: docs/source/get_started/codes.rst [hunk 3]
```diff
@@ -55,69 +55,58 @@ To setup a new code, you execute::
...
-  real execution of the job. It is your responsibility to write proper ``bash`` code!
-  This is intended for code-dependent code. *Remember*
-  *to end the input by pressing* ``<CTRL>+D``.
+At the end of these steps, you will be prompted to edit a script,
+and you will have the opportunity to include ``bash`` commands that will
+be executed *before* running the submission script (after the 
+'pre execution script' lines) and *after* running the submission script 
+(after the 'Post execution script' separator).
+This is intended for code-dependent settings, for instance to load modules or set variables 
+that are needed for the code to be. For example::
```
Comment (borellim) [inline docs/source/get_started/codes.rst]:
```suggestion
that are needed by the code. For example::
```

**Comment by ltalirz** [comment]
Thanks @d-tomerini 
Again, here an example with a yaml file would be great.
You can just copy what was done here:
https://github.com/aiidateam/aiida-tutorials/blob/master/docs/pages/2019_MARVEL_Psik_MaX/sections/calculations.rst#code-setup

**Comment by sphuber** [comment]
Thanks a lot to all

**Pull request merged**